From 771f4123db85652c20d8b40c768528ca74eed4d6 Mon Sep 17 00:00:00 2001
From: Kotresh HR <khiremat@redhat.com>
Date: Fri, 11 Mar 2016 15:07:48 +0530
Subject: [PATCH 34/80] posix: Filter gsyncd stime xattr

Filter gsyncd stime xattr in lookup as well.
The value of stime would be different among
replica bricks and EC bricks. AFR and EC
should not take any action on these as it
could be different.

Change-Id: I0ab4105aa201662c429c3495e8d865449b3554ea
Reviewed on: http://review.gluster.org/13678
BUG: 1294062
Signed-off-by: Kotresh HR <khiremat@redhat.com>
Reviewed-on: http://review.gluster.org/13679
Smoke: Gluster Build System <jenkins@build.gluster.com>
CentOS-regression: Gluster Build System <jenkins@build.gluster.com>
NetBSD-regression: NetBSD Build System <jenkins@build.gluster.org>
Reviewed-by: Pranith Kumar Karampuri <pkarampu@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/70906
Reviewed-by: Aravinda Vishwanathapura Krishna Murthy <avishwan@redhat.com>
Tested-by: Aravinda Vishwanathapura Krishna Murthy <avishwan@redhat.com>
---
 xlators/storage/posix/src/posix-helpers.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/xlators/storage/posix/src/posix-helpers.c b/xlators/storage/posix/src/posix-helpers.c
index c2da976..ba42272 100644
--- a/xlators/storage/posix/src/posix-helpers.c
+++ b/xlators/storage/posix/src/posix-helpers.c
@@ -749,6 +749,9 @@ _handle_list_xattr (dict_t *xattr_req, const char *real_path, int fdnum,
                 if (posix_special_xattr (marker_xattrs, key))
                         goto next;
 
+                if (!fnmatch (GF_XATTR_STIME_PATTERN, key, 0))
+                        goto next;
+
                 if (dict_get (filler->xattr, key))
                         goto next;
 
-- 
1.7.1

